projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -70,13 +84,11 @@ export class IgxFilteringService implements OnDestroy {
             this._filterMenuOverlaySettings.target = filterIconTarget;
             this._filterMenuOverlaySettings.outlet = (this.grid as any).outlet;
 
-            if (this.grid.excelStyleFilteringComponent) {
-                this._componentOverlayId =
-                    this._overlayService.attach(this.grid.excelStyleFilteringComponent.element, this._filterMenuOverlaySettings);
-            } else {
-                this._componentOverlayId =
-                    this._overlayService.attach(classRef, this._filterMenuOverlaySettings, this._moduleRef);
-            }
+            const instance = this.componentInstance;
+            instance.initialize(this.column, this._overlayService, this._componentOverlayId);
```

Comment:
We could remove the `overlayID` from the `initialize` method since it is not initialized before the attach.
```suggestion
            instance.initialize(this.column, this._overlayService);
```